R/flex_dashboard.R

```diff
@@ -263,6 +263,42 @@ flex_dashboard <- function(fig_width = 6.0,
     source_file <<- basename(input)
   }
 
+  #  special handling of 'global' chunk within runtime: shiny
+  if (!is.null(shiny::getDefaultReactiveDomain())) {
+
+    # options hook to (a) capture the chunk options for inspection below and
+    # (b) to ensure that 'global' chunks are not displayed in the document
+    chunk_options <- list()
+    knitr::opts_hooks$set(include = function(options) {
+      chunk_options <<- options
+      if (identical(chunk_options$label, "global")) {
+        options$echo <- FALSE
+        options$include <- FALSE
+      }
+      options
+    })
+
+    # evaluate the 'global' chunk only once, and in the global environment
+    knitr::knit_hooks$set(evaluate = function(code, envir, ...) {
+      if (identical(chunk_options$label, "global")) {
+        # evaluate the global chunk for this source file if it hasn't
+        # been evaluated already
```

Comment:
`code` might be a character vector of length > 1, so let's do `code <- paste(code, collapse = '\n')`